contracts/src/v0.8/ccip/offRamp/OffRampWithMessageTransformer.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {IMessageTransformer} from "../interfaces/IMessageTransformer.sol";
+import {Internal} from "../libraries/Internal.sol";
+import {OffRamp} from "./OffRamp.sol";
+
+/// @notice OffRamp that uses a message transformer to transform messages before execution
+contract OffRampWithMessageTransformer is OffRamp {
+  address internal s_messageTransformer;
+
+  constructor(
+    StaticConfig memory staticConfig,
+    DynamicConfig memory dynamicConfig,
+    SourceChainConfigArgs[] memory sourceChainConfigs,
+    address messageTransformerAddr
+  ) OffRamp(staticConfig, dynamicConfig, sourceChainConfigs) {
+    if (address(messageTransformerAddr) == address(0)) {
+      revert ZeroAddressNotAllowed();
+    }
+    s_messageTransformer = messageTransformerAddr;
+  }
+
+  function getMessageTransformer() external view returns (address) {
```

Comment:
All functions for every CCIP contract have natspec comments, please add them to all (non-test) contracts in this PR